camply/cli.py

```diff
@@ -27,8 +29,15 @@
 @click.option(
     "--debug/--no-debug", default=False, help="Enable extra debugging output."
 )
+@click.option(
```

Comment:
👏 👏 @acaloiaro thank you so much - I'm starting to look at this on my end. I've got some big life events this week (🤵 💍 👰‍♀️ ) and so review will be delayed. 

One question on the `--provider` change:
- I love moving `--provider` to the top level of the CLI. One question though, does this mean that passing it in after `--campsites` will break? I really like options like `--profile` in the AWS CLI where you can just throw it in wherever. I'm wondering if Click gives us any kind of flexibility on those universal options like `--provider`

Reply:
In its current implementation, I think it does break that behavior, but let me see if click will allow the alternate behavior at the same time.

Reply:
BTW I still need to add some tests and manually test more things, like notifications and configuration file support, before this PR will actually be ready for consideration. I just wanted to get another set of eyes on it for now.

Reply:
Hey @acaloiaro following up here. I've actually moved `--provider` and `--debug` to the top level and lower level CLI namespaces via some Click Contexts on Release [0.7.0](https://github.com/juftin/camply/releases/tag/v0.7.0).

I've also added a [Discord Server](https://discord.gg/qZDr78kKvB) - it's pretty much empty but I wanted to make sure there's an easy place to collaborate for features like yours. Is there anything I can do to help you along on this right now?

Reply:
Cool! I'm taking a look at this PR again now and will try to get it polished up.